easyrest/views/user_controller.py

```diff
@@ -50,3 +55,58 @@ def sign_up(request):
     except IntegrityError:
         database.rollback()
         raise HTTPForbidden("User already exists!")
+
+
+@view_config(route_name='users_list', renderer='json', request_method='GET')
+def get_users_list(request):
+    """This function is intended to display a list of users
+       depending on the role id.
+
+       This function processes the route /users/{role_id}
+       and tries to create a list of users depending on the identifier
+       that is extracted from the parameter {role_id}.
+
+        :param request: GET request
+        :raise HTTPForbidden: If the token is not found
+        :return: Users list, if the user is allowed to view users with the specified role:
+                    {
+                      "message": "Users with role 'role_name'",
+                        "data": [
+                          {
+                            "phone_number": "user_phone_number",
+                            "name": "user_name",
+                            "is_active": is_active_state,
+                            "id": user_id,
+                            "birth_date": "user_birth_date",
+                            "email": "user_email"
+                          }
+                        ],
+                      "success": true,
+                      "error": null
+                    }
+
+                 If the user is not allowed to view the list of users with the specified role:
+                    {
+                      "message": null,
+                      "data": [],
+                      "success": false,
+                      "error": "Action not allowed"
+                    }
+
+    """
+    log = logging.getLogger(__name__)
+    try:
+        current_user = request.token.user
+    except AttributeError as ae:
+        log.error(ae.message)
+        raise HTTPForbidden('Need token for access')
```

Comment:
Why you do not use auth.restrict_access(user_types=['Client', 'Owner', 'Moderator', 'Admin'....])?

Reply:
You're right. In this case, it is better to use the restrict_access.

Reply:
By the way, access restriction can only be used as a decorator, is not it?
`restrict_access` will not raise an exception if I use it in the function body as `restrict_access(user_types=['Client', 'Owner', 'Moderator', 'Admin'])`, am I right?

For example:
```
from ..auth import restrict_access

def get_users_list(request):
    restrict_access(user_types=['Client', 'Owner', 'Moderator', 'Admin'])
    # There must be an exception HTTPForbidden("Need token for access")
    # if the user is not registered but the script continues to be executed further  
    
    derivable_role_id = int(request.matchdict['role_id'])
    role = request.dbsession.query(UserRole).get(derivable_role_id)
    if role is None:
        raise HTTPNotFound(request.path)
```

Reply:
Why you cant use it as a decorator above def get_users_list(request):?

Reply:
So that's what I did in this case. See `eee54c5`